Edge tools for vs code  || node and npm

Ensure `node` and `npm` are installed to enable automatically reporting issues in source files pertaining to accessibility, compatibility, security, and more.

what is this? how can i solve this?

Saw the same error. I don't see a prompt to check the output of the extension, but a detailed error is shown in there.

<details>
<summary>Output of Microsoft Edge Tools</summary>

```
Trying to load shared version
Found hint at c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\index.js
Error loading shared "hint" package
Error: Cannot find module 'content-type'
Require stack:
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-network\dist\src\is-html-document.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-network\dist\src\index.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-json\dist\src\final-config.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-json\dist\src\index.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\config\config-hints.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\engine.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\index.js
- c:\Users\user\.vscode\extensions\ms-edgedevtools.vscode-edge-devtools-2.1.1\node_modules\vscode-webhint\dist\src\server.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:987:15)
    at Module._load (node:internal/modules/cjs/loader:832:27)
    at c._load (node:electron/js2c/asar_bundle:5:13343)
    at Module.require (node:internal/modules/cjs/loader:1059:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-network\dist\src\is-html-document.js:4:24)
    at Module._compile (node:internal/modules/cjs/loader:1165:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1220:10)
    at Module.load (node:internal/modules/cjs/loader:1035:32)
    at Module._load (node:internal/modules/cjs/loader:876:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-network\\dist\\src\\is-html-document.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-network\\dist\\src\\index.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-json\\dist\\src\\final-config.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-json\\dist\\src\\index.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\hint\\dist\\src\\lib\\config\\config-hints.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\hint\\dist\\src\\lib\\engine.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\hint\\dist\\src\\lib\\index.js',
    'c:\\Users\\user\\.vscode\\extensions\\ms-edgedevtools.vscode-edge-devtools-2.1.1\\node_modules\\vscode-webhint\\dist\\src\\server.js'
  ]
}
Unable to load shared webhint instance Error: Cannot find module 'content-type'
Require stack:
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-network\dist\src\is-html-document.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-network\dist\src\index.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-json\dist\src\final-config.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-json\dist\src\index.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\config\config-hints.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\engine.js
- c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\index.js
- c:\Users\user\.vscode\extensions\ms-edgedevtools.vscode-edge-devtools-2.1.1\node_modules\vscode-webhint\dist\src\server.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:987:15)
    at Module._load (node:internal/modules/cjs/loader:832:27)
    at c._load (node:electron/js2c/asar_bundle:5:13343)
    at Module.require (node:internal/modules/cjs/loader:1059:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\@hint\utils-network\dist\src\is-html-document.js:4:24)
    at Module._compile (node:internal/modules/cjs/loader:1165:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1220:10)
    at Module.load (node:internal/modules/cjs/loader:1035:32)
    at Module._load (node:internal/modules/cjs/loader:876:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-network\\dist\\src\\is-html-document.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-network\\dist\\src\\index.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-json\\dist\\src\\final-config.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\@hint\\utils-json\\dist\\src\\index.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\hint\\dist\\src\\lib\\config\\config-hints.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\hint\\dist\\src\\lib\\engine.js',
    'c:\\Users\\user\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-edgedevtools.vscode-edge-devtools\\node_modules\\hint\\dist\\src\\lib\\index.js',
    'c:\\Users\\user\\.vscode\\extensions\\ms-edgedevtools.vscode-edge-devtools-2.1.1\\node_modules\\vscode-webhint\\dist\\src\\server.js'
  ]
}
Installing shared version of "hint"
Last check for "hint" updates was less than 24 hours ago, skipping
Found hint at c:\Users\user\AppData\Roaming\Code\User\globalStorage\ms-edgedevtools.vscode-edge-devtools\node_modules\hint\dist\src\lib\index.js
```

</details>

I fixed by:

- Open a terminal and navigate to `C:/Users/[username]/AppData/Roaming/Code/User/globalStorage/ms-edgedevtools.vscode-edge-devtools` (depends on OS youre using)
- Rename `node_modules` to `node_modules_` or something
- Run `npm i -D`
- Reload vscode

--- a/package.json
+++ b/package.json
@@ -278,6 +278,11 @@
                     "type": "boolean",
                     "default": true,
                     "description": "Enable feedback from webhint on source files to improve accessibility, compatibility, security and more."
+                },
+                "vscode-edge-devtools.webhintInstallNotification": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "Turn off notification for webhint installation failures."
                 }
             }
         },

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -346,14 +346,20 @@ function startWebhint(context: vscode.ExtensionContext): void {
     void client.onReady().then(() => {
         // Listen for notification that the webhint install failed.
         const installFailedNotification: typeof installFailed = 'vscode-webhint/install-failed';
-        client.onNotification(installFailedNotification, () => {
-            const message = 'Ensure `node` and `npm` are installed to enable automatically reporting issues in source files pertaining to accessibility, compatibility, security, and more.';
-            void vscode.window.showInformationMessage(message, 'OK', 'Disable').then(button => {
-                if (button === 'Disable') {
-                    void vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).update('webhint', false, vscode.ConfigurationTarget.Global);
-                }
+        const disableNotication = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('webhintInstallNotification');
+        if (!disableNotication) {
+            client.onNotification(installFailedNotification, () => {
+                const message = 'Ensure `node` and `npm` are installed to enable automatically reporting issues in source files pertaining to accessibility, compatibility, security, and more.';
+                void vscode.window.showInformationMessage(message, 'Remind me Later', 'Disable Extension', 'Don\'t show again').then(button => {
+                    if (button === 'Disable Extension') {
+                        void vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).update('webhint', false, vscode.ConfigurationTarget.Global);
+                    }
+                    if (button === 'Don\'t show again') {
+                        void vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).update('webhintInstallNotification', true, vscode.ConfigurationTarget.Global);
+                    }
+                });
             });
-        });
+        }
         // Listen for requests to show the output panel for this extension.
         const showOutputNotification: typeof showOutput = 'vscode-webhint/show-output';
         client.onNotification(showOutputNotification, () => {

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -346,13 +346,13 @@ function startWebhint(context: vscode.ExtensionContext): void {
     void client.onReady().then(() => {
         // Listen for notification that the webhint install failed.
         const installFailedNotification: typeof installFailed = 'vscode-webhint/install-failed';
-        const disableInstallNotification = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('webhintInstallNotification');
+        const disableInstallFailedNotification = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('webhintInstallNotification');
         client.onNotification(installFailedNotification, () => {
             if (!telemetryReporter) {
                 telemetryReporter = createTelemetryReporter(context);
             }
             telemetryReporter.sendTelemetryEvent('user/webhint/install-failed');
-            if (!disableInstallNotification) {
+            if (!disableInstallFailedNotification) {
                 const message = 'Ensure `node` and `npm` are installed to enable automatically reporting issues in source files pertaining to accessibility, compatibility, security, and more.';
                 void vscode.window.showInformationMessage(message, 'Remind me Later', 'Don\'t show again', 'Disable Extension').then(button => {
                     if (button === 'Disable Extension') {

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -346,11 +346,15 @@ function startWebhint(context: vscode.ExtensionContext): void {
     void client.onReady().then(() => {
         // Listen for notification that the webhint install failed.
         const installFailedNotification: typeof installFailed = 'vscode-webhint/install-failed';
-        const disableNotication = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('webhintInstallNotification');
-        if (!disableNotication) {
-            client.onNotification(installFailedNotification, () => {
+        const disableInstallNotification = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('webhintInstallNotification');
+        client.onNotification(installFailedNotification, () => {
+            if (!telemetryReporter) {
+                telemetryReporter = createTelemetryReporter(context);
+            }
+            telemetryReporter.sendTelemetryEvent('user/webhint/install-failed');
+            if (!disableInstallNotification) {
                 const message = 'Ensure `node` and `npm` are installed to enable automatically reporting issues in source files pertaining to accessibility, compatibility, security, and more.';
-                void vscode.window.showInformationMessage(message, 'Remind me Later', 'Disable Extension', 'Don\'t show again').then(button => {
+                void vscode.window.showInformationMessage(message, 'Remind me Later', 'Don\'t show again', 'Disable Extension').then(button => {
                     if (button === 'Disable Extension') {
                         void vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).update('webhint', false, vscode.ConfigurationTarget.Global);
                     }
@@ -358,8 +362,8 @@ function startWebhint(context: vscode.ExtensionContext): void {
                         void vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).update('webhintInstallNotification', true, vscode.ConfigurationTarget.Global);
                     }
                 });
-            });
-        }
+            }
+        });
         // Listen for requests to show the output panel for this extension.
         const showOutputNotification: typeof showOutput = 'vscode-webhint/show-output';
         client.onNotification(showOutputNotification, () => {

